scrapy/resolver.py

```diff
@@ -72,10 +69,8 @@ def from_crawler(cls, crawler, reactor):
             cache_size = 0
         return cls(reactor, cache_size)
 
-    @classmethod
-    def install_on_reactor(cls, reactor, crawler):
-        resolver = create_instance(cls, None, crawler, reactor=reactor)
-        reactor.installNameResolver(resolver)
+    def install_on_reactor(self, reactor):
```

Comment:
I wonder if a better API would be not to require reactor argument - it is already passed to `__init__`.

Reply:
hm makes sense, we can keep a reference to the reactor and use it internally.